packages/algob/src/lib/tx.ts

```diff
@@ -141,6 +143,42 @@ async function sendAndWait (
   return await deployer.waitForConfirmation(txInfo.txId);
 }
 
+/**
+ * Make transaction parameters and update ASA and SSC params
+ * @param deployer Deployer object
+ * @param t Execution parameters
+ * @param index index of current execParam
+ * @param txIdxMap Map for index to name
+ */
+async function mkTx (
+  deployer: Deployer,
+  txn: rtypes.ExecParams,
+  index: number,
+  txIdxMap: Map<number, string>
+): Promise<Transaction> {
+  switch (txn.type) {
+    case TransactionType.DeployASA: {
+      deployer.assertNoAsset(txn.asaName);
```

Comment:
just to confirm - this is where the tx will fail if i used `deployASA`/`deploySSC` in *RUN* mode, right? With error message of cannot use assertNoAsset in run mode.. ?

Reply:
yes